pkg/pillar/devicenetwork/dnc.go

```diff
@@ -430,44 +418,57 @@ func VerifyDevicePortConfig(ctx *DeviceNetworkContext) {
 					res.String(), ctx.DevicePortConfigList.PortConfigList[ctx.NextDPCIndex].Key,
 					pending.PendDPC.Key)
 			}
-			passed = true
-			if ctx.NextDPCIndex == 0 {
-				log.Functionf("VerifyDevicePortConfig: Working DPC configuration found "+
-					"at index %d in DPC list",
-					ctx.NextDPCIndex)
-			} else {
-				log.Warnf("VerifyDevicePortConfig: Working DPC configuration found "+
-					"at index %d in DPC list",
-					ctx.NextDPCIndex)
-				if ctx.NetworkTestBetterInterval != 0 {
-					// Look for a better choice in a while
-					duration := time.Duration(ctx.NetworkTestBetterInterval) * time.Second
-					ctx.NetworkTestBetterTimer = time.NewTimer(duration)
-				}
-			}
+			endloop = true
+			log.Functionf("VerifyDevicePortConfig: Working DPC configuration found "+
+				"at index %d in DPC list", ctx.NextDPCIndex)
 		}
 	}
-	// Found a working one
-	ctx.DevicePortConfigList.CurrentIndex = ctx.NextDPCIndex
-	*ctx.DevicePortConfig = pending.PendDPC
-	*ctx.DeviceNetworkStatus = pending.PendDNS
-	ctx.DeviceNetworkStatus.Testing = false
-	*ctx.DevicePortConfigList = compressAndPublishDevicePortConfigList(ctx)
-	DoDNSUpdate(ctx)
 
+	if ctx.NextDPCIndex != 0 {
+		log.Warnf("VerifyDevicePortConfig: Working with DPC configuration found "+
+			"at index %d in DPC list",
+			ctx.NextDPCIndex)
+		if ctx.NetworkTestBetterInterval != 0 {
+			// Look for a better choice in a while
+			duration := time.Duration(ctx.NetworkTestBetterInterval) * time.Second
+			ctx.NetworkTestBetterTimer = time.NewTimer(duration)
+			log.Warnf("VerifyDevicePortConfig: Kick started NetworkTestBetterTimer " +
+				"to try and get back to DPC at Index 0")
+		} else {
+			log.Warnf("VerifyDevicePortConfig: Did not start NetworkTestBetterTimer " +
+				"since timer interval is configured to be zero")
+		}
+	}
 	pending.Inprogress = false
 
-	// Did we get a new at index zero?
-	if ctx.DevicePortConfigList.PortConfigList[0].IsDPCUntested() {
-		log.Warn("VerifyDevicePortConfig DPC_SUCCESS: New DPC arrived " +
-			"or a old working DPC moved up to top of DPC list while network testing " +
-			"was in progress. Restarting DPC verification.")
-		RestartVerify(ctx, "VerifyDevicePortConfig DPC_SUCCESS")
-		return
-	}
+	switch res {
+	case types.DPC_SUCCESS, types.DPC_REMOTE_WAIT:
```

Comment:
Do we even need to check res except for setting ctx.CloudConnectivityWorks = true
Otherwise the code is the same (and IsDPCUntested makes sense even if there was a failure).

Reply:
We do not execute the below code when res == DPC_FAIL

Reply:
Ok. I think we can re-start the verification when IsDPCUntested is true even for DPC_FAIL. WIll do that